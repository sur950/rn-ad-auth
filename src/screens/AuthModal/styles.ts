import { StyleSheet, Dimensions } from 'react-native';

const { height, width } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: height - 20,
    width: width,
  },
  centeredView: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: 'green',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    borderRadius: 20,
    padding: 10,
    elevation: 2,
  },
  buttonOpen: {
    backgroundColor: '#F194FF',
  },
  buttonClose: {
    backgroundColor: '#2196F3',
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalText: {
    marginBottom: 15,
    textAlign: 'center',
  },
});

export default styles;
