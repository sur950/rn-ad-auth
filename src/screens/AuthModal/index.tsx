import React, { useState } from 'react';
import { Modal, View } from 'react-native';
import AuthView from '../AuthView';
import type { AuthViewProps } from '../../types';
import styles from './styles';

const AuthModal = (props: AuthViewProps) => {
  const { open } = props;
  const [modalVisibility, setModalVisibility] = useState<boolean>(open);

  React.useEffect(() => {
    setModalVisibility(open);
  }, [open]);

  return (
    <View style={styles.container}>
      <Modal
        presentationStyle="fullScreen"
        supportedOrientations={['portrait']}
        animationType="slide"
        visible={modalVisibility}
      >
        <AuthView
          {...props}
          closeModal={() => {
            setModalVisibility(false);
          }}
        />
      </Modal>
    </View>
  );
};

export default AuthModal;
