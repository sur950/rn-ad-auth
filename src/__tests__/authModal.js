// __tests__ for AuthModal Widget

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AuthModal from '../screens/AuthModal';

describe('Authview test', () => {
  const config = {
    tenant_id: 'e38a8b10-6073-4eb0-93bf-8b729e9ef139',
    client_id: '5a19138e-1cf2-4127-bb31-9fffedcea6fb',
    scope: 'openid',
  };

  const onSuccess = (response) => {
    console.log('response', response);
  };

  const onFailure = (error) => {
    console.error('error', error);
  };

  const onClose = () => {
    console.log('onClose received');
  };

  it(`should match the AuthModal snapshot`, () => {
    const tree = renderer
      .create(
        <AuthModal
          config={config}
          action={'login'}
          open={true}
          onClose={onClose}
          onSuccess={onSuccess}
          onFailure={onFailure}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
