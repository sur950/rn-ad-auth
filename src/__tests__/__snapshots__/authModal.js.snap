// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authview test should match the AuthModal snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1314,
      "width": 750,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    presentationStyle="fullScreen"
    supportedOrientations={
      Array [
        "portrait",
      ]
    }
    visible={true}
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#f8f7f8",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f8f8",
            "borderBottomColor": "#d8d8d8",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 44,
            "paddingHorizontal": 20,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#007aff",
                "fontSize": 18,
                "fontWeight": "400",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 12,
                "width": 12,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "marginLeft": 5,
              }
            }
          >
            https://login.microsoftonline.com
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RNCWebView
          cacheEnabled={true}
          injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
          injectedJavaScriptForMainFrameOnly={true}
          javaScriptEnabled={true}
          messagingEnabled={true}
          onContentProcessDidTerminate={[Function]}
          onHttpError={[Function]}
          onLoadingError={[Function]}
          onLoadingFinish={[Function]}
          onLoadingProgress={[Function]}
          onLoadingStart={[Function]}
          onMessage={[Function]}
          onShouldStartLoadWithRequest={[Function]}
          source={
            Object {
              "uri": "https://login.microsoftonline.com/e38a8b10-6073-4eb0-93bf-8b729e9ef139/oauth2/v2.0/authorize/?tenant_id=e38a8b10-6073-4eb0-93bf-8b729e9ef139&client_id=5a19138e-1cf2-4127-bb31-9fffedcea6fb&scope=openid&response_type=code&response_mode=query&state=12345&nonce=678910&redirect_uri=https%3A%2F%2Fazure-ad-login-3606b.web.app%2F",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              Object {
                "backgroundColor": "#f8f7f8",
                "flex": 1,
              },
            ]
          }
          textInteractionEnabled={true}
          useSharedProcessPool={true}
        />
      </View>
    </RCTSafeAreaView>
  </Modal>
</View>
`;
